R/utils.R

```diff
@@ -327,4 +335,66 @@ valid_seq <- function(params) {
   } else {
     return(FALSE)
   }
-}
\ No newline at end of file
+}
+
+resolve_prefix <- function(prefix, environment_var) {
+  if (!(is.null(prefix))) {
+    assertthat::assert_that(is.character(prefix))
+    
+    return(prefix)
+  } else {
+    prefix_env <- Sys.getenv(environment_var)
+    if (prefix_env != "") {
+      return(prefix_env)
+    } else {
+      return("/")
+    }
+  }
+}
+
+get_package_mapping <- function(script_path, url_package, dependencies) {
+  package_map <- vapply(dependencies, function(x) {
+    if (x$name %in% c('react', 'react-dom')) {
+      x$name <- 'dash_renderer'
+    }
+    dep_path <- paste(x$src$file, 
+                      x$script,
+                      sep = "/")
```

Comment:
is this a file path? if so, should this be platform dependent or is there a platform independent way to construct platform paths? (on windows, it's `\`)

Reply:
Right, good question; I can't say with 100% certainty, but I think it might be OK here because of the way `system.file` is written:

https://github.com/wch/r-source/blob/5a156a0865362bb8381dcd69ac335f5174a4f60c/src/library/base/R/files.R#L201

...the line above calls `file.path`, which is platform-agnostic. From https://www.rdocumentation.org/packages/base/versions/3.5.1/topics/file.path:

>_Construct the path to a file from components in a platform-independent way._

I can test the code on my Windows VM and verify that this is the case.

Reply:
Got it - let's use `file.path` here and anywhere that we're constructing file paths. That way, we're not dependent on other functions downstream handling our non-platform-dependent file strings correctly.

In python we do a similar thing with `os.path.join`

Reply:
can we get into the habit of using `file.path` here?

Reply:
Or, since you are replacing `/` below, maybe you don't want to use `file.path` here because that could return `/` on windows? If that's the case, could add a comment to elaborate?

Reply:
fixed in 3a61d023c1603b9d6d1f6303351b620cbb5262ac